test/success/num.prl

```diff
@@ -96,4 +96,45 @@ Thm NatSymm : [
       [i=1 {_} ,a])
 ].
 
-Print NatSymm.
\ No newline at end of file
+Print NatSymm.
+
+Thm IntPred : [
+  (-> int int)
+] by [
+  lam a. fresh a', junk -> elim a;
+  [ `-1, fresh n, junk -> elim a'; [`0, `(succ ,n)]
```

Comment:
So, if we have subtyping, then `fresh n, junk -> elim a'; [exact 0, exact (succ ,n)]` can be replaced by `use a'`.